### DISCUSSION TITLE
Rate rule error should be checked at invoice creation

### DISCUSSION THREAD

**Discussion by jacobburrell** [btcpayserver/btcpayserver#4188] [Ideas / Feature Requests]
Currently, in [Payment Requests](https://docs.btcpayserver.org/PaymentRequests/), it seems missing exchange rates will check for the exchange rate only upon an attempt to pay the payment request (and thus create a [invoice](https://docs.btcpayserver.org/Invoices/).)

This UX presented an unnecessary issue while making an invoice the other day as the Payment Request was created successfully, and the user creating the issue shared the payment invoice. Due to the _rate rule error_, the Payment Request was unpayable and the request had to be returned to the issuer to be fixed.

The UX probably should present the error either at invoice creation where the currency is selected, and/or when  **Default Currency Pairs** are selected, in this case USD_MXN is able to be set and saved as a DCP, but fails when manually set as a "Test".

This would prevent unpayable invoices due to a simple rate error rule. 

You also might be able to include fallbacks (e.g. use USD instead of MXN if MXN fails, or use another source for USD:MXN rates as a **proxy rate**.

Example error message when attempting to pay invoice:
`Error creating invoice: Warning: You have payment methods configured but none of them match any of the requested payment methods or the rate is not available. See logs below: 10/04/2022 04:34:53:Info Creation of invoice starting 10/04/2022 04:34:53:Info BTC_MXN: The rating rule is kraken(BTC_MXN) 10/04/2022 04:34:53:Info BTC_MXN: The evaluated rating rule is ERR_RATE_UNAVAILABLE(kraken, BTC_MXN) 10/04/2022 04:34:53:Error BTC_MXN: Rate rule error (RateUnavailable)`

**Comment by almostheaven1863** [comment]
Why would you get the above error? I am getting this error when i made a payment request. I have not changed any settings, so I am confused why it is no longer working.

**Reply by jacobburrell** [reply]
Rate lookup fails and thus the entire invoice and thus request also fails.

**Comment by jacobburrell** [comment]
@pavlenex can I make a PR for this?

Also, can we set better rate defaults? 

E.g. if store currency is MXN
<img width="315" height="134" alt="image" src="https://github.com/user-attachments/assets/eaa658f0-8a12-4e45-9cd9-d136d24fce13" />

By **default** the rate source is Kraken which does not have the store's fiat and BTC's rate available. 
<img width="888" height="280" alt="image" src="https://github.com/user-attachments/assets/a454eb98-97e4-4be0-b585-71c1a74a8649" />

<img width="980" height="240" alt="image" src="https://github.com/user-attachments/assets/abff1254-8816-4554-a697-618445b39907" />

While it might also be a good idea to set the primary rate source by default with a provider that has your store's fiat, I would like to set Fallback Rate Sources **on by default** with several exchanges selected.
